sopel/modules/remind.py

```diff
@@ -8,80 +8,150 @@
-    return os.path.join(self.config.core.homedir, name)
+    :param str filename: absolute path to the remind database file
+    :return: a :class:`dict` of reminders stored by timestamp
+    :rtype: dict
 
+    The remind database is a plain text file (utf-8 encoded) with tab-separated
+    columns of data: time, channel, nick, and message. This function reads this
+    file and outputs a :class:`dict` where keys are the timestamps of the
+    reminders, and values are list of 3-value tuple of reminder data:
+    ``(channel, nick, message)``.
+
+    .. note::
+
+        This function ignores microseconds from the timestamp, if any, meaning
+        that ``523549800.245`` will be read as ``523549800``.
+
+    .. note::
+
+        If ``filename`` is not an existing file, this function returns an
+        empty :class:`dict`.
+
+    """
+    if not os.path.isfile(filename):
+        # no file to read
+        return {}
 
-def load_database(name):
     data = {}
-    if os.path.isfile(name):
-        f = codecs.open(name, 'r', encoding='utf-8')
-        for line in f:
-            unixtime, channel, nick, message = line.split('\t')
+    with codecs.open(filename, 'r', encoding='utf-8') as database:
+        for line in database:
+            unixtime, channel, nick, message = line.split('\t', 3)
             message = message.rstrip('\n')
-            t = int(float(unixtime))  # WTFs going on here?
+            timestamp = int(float(unixtime))  # ignore microseconds
             reminder = (channel, nick, message)
             try:
-                data[t].append(reminder)
+                data[timestamp].append(reminder)
             except KeyError:
-                data[t] = [reminder]
-        f.close()
+                data[timestamp] = [reminder]
     return data
 
 
-def dump_database(name, data):
-    f = codecs.open(name, 'w', encoding='utf-8')
-    for unixtime, reminders in sopel.tools.iteritems(data):
-        for channel, nick, message in reminders:
-            f.write('%s\t%s\t%s\t%s\n' % (unixtime, channel, nick, message))
-    f.close()
+def dump_database(filename, data):
+    """Dump the remind database into a file
+
+    :param str filename: absolute path to the remind database file
+    :param dict data: remind database to dump
+
+    The remind database is dumped into a plain text file (utf-8 encoded) as
+    tab-separated columns of data: unixtime, channel, nick, and message.
+
+    If the file does not exist, it is created.
+    """
+    with codecs.open(filename, 'w', encoding='utf-8') as database:
+        for unixtime, reminders in tools.iteritems(data):
+            for channel, nick, message in reminders:
+                line = '%s\t%s\t%s\t%s\n' % (unixtime, channel, nick, message)
+                database.write(line)
+
+
+def create_reminder(bot, trigger, duration, message, timezone):
+    """Create a reminder into the ``bot`` and reply to the sender"""
```

Comment:
"into the bot"? :thinking:

Reply:
Did I really write that? Even by my poor standard, it feels wrong.

Reply:
Well yes. I did. But why?